### ISSUE TITLE
Implement TCGC's SdkPackage types

### ISSUE THREAD

**Issue by iscai-msft** [Azure/autorest.csharp#4534]
Language emitters should work to move entirely to TCGC's new ecosystem. The entry point for the ecosystem is accessing .sdkPackage on the SdkContext returned from tcgc
```[tasklist]
### Tasks
- [ ] https://github.com/Azure/autorest.csharp/issues/4747
```

**Comment by archerzz** [comment]
I believe this is actually a duplicate of #3994

**Comment by archerzz** [comment]
close since it's resolved in #4300
